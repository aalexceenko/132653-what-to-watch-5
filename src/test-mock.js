import {getFilmRank} from "./utils";

export const VISIBLE_FILMS_COUNT = 8;
export const ID = 1;
export const VIDEO_CURRENT_TIME = 0;
export const PROGRESS_POSITION = 0;
export const AUTHORIZATION_STATUS = `AUTH`;

export const IS_VIDEO = true;
export const IS_PLAYING = true;
export const IS_IN_MY_LIST = false;
export const TAB = `details`;
export const PREVIEW = `img/aviator.jpg`;
export const PREVIEW_VIDEO = `https://upload.wikimedia.org/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`;
export const TITLE = `Bohemian Rhapsody`;
export const ACTIVE_GENRE = `Comedy`;
export const GENRES = [`Comedy`, `Crime`, `Documentary`, `Drama`, `Horror`, `Kids & Family`];
export const MATCH = {
  params: {
    id: 1
  }
};

export const FILMS = [
  {
    description: `A young man who was sentenced to seven years`,
    rating: 3.6,
    director: `Nicolas Winding Refn`,
    genre: `Action`,
    actors: [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
    id: 1,
    title: `Bronson`,
    poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
    previewVideo: `https://upload.wikimedia.org/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    trailer: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    votes: 109661,
    runtime: 92,
    year: 2008,
    ratingText: getFilmRank(3.6),
    myList: false,
    comments: false,
  },
  {
    description: `A young man who was`,
    rating: 3.6,
    director: `Nicolas Winding Refn`,
    genre: `Action`,
    actors: [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
    id: 2,
    title: `Bronson`,
    poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
    backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
    previewVideo: `https://upload.wikimedia.org/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    trailer: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    votes: 109661,
    runtime: 92,
    year: 2008,
    ratingText: getFilmRank(3.6),
    myList: false,
    comments: false,
  },
];

export const FILM = {
  description: `A young man who was sentenced to seven years`,
  rating: 3.6,
  director: `Nicolas Winding Refn`,
  genre: `Action`,
  id: 3,
  title: `Bronson`,
  actors: [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
  poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
  previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
  backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
  previewVideo: `https://upload.wikimedia.org/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  trailer: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  votes: 109661,
  runtime: 92,
  year: 2008,
  ratingText: getFilmRank(3.6),
  myList: false,
  comments: false,
};

export const REVIEWS = [
  {
    rating: 3.3,
    date: `2020-10-09T13:38:44.769Z`,
    user: {
      name: `Christina`,
      id: 1,
    },
    comment: `This film is an experience and i has already seen it 4 times`,
  },
  {
    rating: 3.3,
    date: `2020-10-09T13:38:44.769Z`,
    user: {
      name: `Christina`,
      id: 2,
    },
    comment: `This film is an experience and i has already seen it 4 times`,
  }
];
