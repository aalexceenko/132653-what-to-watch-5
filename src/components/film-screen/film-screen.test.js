import React from "react";
import renderer from "react-test-renderer";
import FilmScreen from "./film-screen";
import {FILMS, MATCH} from "../../test-mock";
import {BrowserRouter} from 'react-router-dom';

jest.mock(`../user-block/user-block`, () => `UserBlock`);
jest.mock(`../my-list-button/my-list-button`, () => `MyListButton`);
it(`Should FilmScreen render correctly`, () => {
  const tree = renderer
    .create(
        <BrowserRouter >
          <FilmScreen
            films={FILMS}
            onFilmCardClick={() => {}}
            match={MATCH}
            handleButtonPlayVideo={() => {}}
          />
        </BrowserRouter>

    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
