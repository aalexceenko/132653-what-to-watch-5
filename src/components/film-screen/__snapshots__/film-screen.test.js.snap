// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should FilmScreen render correctly 1`] = `
Array [
  <section
    className="movie-card movie-card--full"
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="Bronson"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header movie-card__head"
      >
        <div
          className="logo"
        >
          <a
            className="logo__link "
            href="/"
            onClick={[Function]}
          >
            <span
              className="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              className="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              className="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <UserBlock />
      </header>
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Bronson
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Action
            </span>
            <span
              className="movie-card__year"
            >
              2008
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <MyListButton
              film={
                Object {
                  "actors": Array [
                    "Tom Hardy",
                    "Kelly Adams",
                    "Luing Andrews",
                  ],
                  "backgroundImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg",
                  "comments": false,
                  "description": "A young man who was sentenced to seven years",
                  "director": "Nicolas Winding Refn",
                  "genre": "Action",
                  "id": 1,
                  "myList": false,
                  "poster": "https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg",
                  "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg",
                  "previewVideo": "https://upload.wikimedia.org/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
                  "rating": 3.6,
                  "ratingText": "Normal",
                  "runtime": 92,
                  "title": "Bronson",
                  "trailer": "http://media.xiph.org/mango/tears_of_steel_1080p.webm",
                  "votes": 109661,
                  "year": 2008,
                }
              }
            />
            <a
              className="btn movie-card__button"
              href="/films/1/review"
              onClick={[Function]}
            >
              Add review
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="Bronson"
            height="327"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <nav
            className="movie-nav movie-card__nav"
          >
            <ul
              className="movie-nav__list"
            >
              <li
                className="movie-nav__item movie-nav__item--active"
              >
                <a
                  className="movie-nav__link"
                  onClick={[Function]}
                >
                  Overview
                </a>
              </li>
              <li
                className="movie-nav__item "
              >
                <a
                  className="movie-nav__link"
                  onClick={[Function]}
                >
                  Details
                </a>
              </li>
              <li
                className="movie-nav__item "
              >
                <a
                  className="movie-nav__link"
                  onClick={[Function]}
                >
                  Reviews
                </a>
              </li>
            </ul>
          </nav>
          <div
            className="movie-rating"
          >
            <div
              className="movie-rating__score"
            >
              3.6
               
            </div>
            <p
              className="movie-rating__meta"
            >
              <span
                className="movie-rating__level"
              >
                Normal
                 
              </span>
              <span
                className="movie-rating__count"
              >
                109661
                 ratings
              </span>
            </p>
          </div>
          <div
            className="movie-card__text"
          >
            <p>
              A young man who was sentenced to seven years
            </p>
            <p
              className="movie-card__director"
            >
              <strong>
                Director: 
                Nicolas Winding Refn
                 
              </strong>
            </p>
            <p
              className="movie-card__starring"
            >
              <strong>
                Starring: 
                Tom Hardy, Kelly Adams, Luing Andrews
                 
              </strong>
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog catalog--like-this"
    >
      <h2
        className="catalog__title"
      >
        More like this
      </h2>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <img
              alt="Bronson"
              height="175"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/2"
              onClick={[Function]}
            >
              Bronson
            </a>
          </h3>
        </article>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
